lib/internal/main/run_main_module.js

```diff
@@ -3,11 +3,36 @@
 const {
   prepareMainThreadExecution
 } = require('internal/bootstrap/pre_execution');
+const {
+  codes: {
+    ERR_SCRIPT_EXECUTION_INTERRUPTED
+  },
+  hideStackFrames
+} = require('internal/errors');
+const { SigintWatchdog } = require('internal/watchdog');
 
 prepareMainThreadExecution(true);
 
 const CJSModule = require('internal/modules/cjs/loader').Module;
 
+const watchdog = new SigintWatchdog();
```

Comment:
This always remains alive until the end of the process, right? (Just making sure, because otherwise there’s a `.close()` call that should be added somewhere.)

Also, maybe setup code should be part of `prepareMainThreadExecution()`, so that it is not only in effect when running `node <scriptname>`?

Reply:
Yes, it would last until the end of the process. I'm just thinking about attaching it to process instance to make this explicit and enable works like 27406.

Reply:
I think that might open a whole bunch of issues, so maybe not in this PR?

Reply:
So I'm just gonna attach it to the global process instance and leave it there, would this be good?

Reply:
The thing about attaching it to the `process` object is that it effectively makes `SigintWatchdog` public API, and I’d be a bit reluctant to do that. I think the code here is fine as it is, I’m only really concerned about under which scenarios we should enable this feature